### PR TITLE
List and other example updates

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Update List, Link, and MarqueeSelection examples to use CSS modules instead of global class names. The examples still need a full css-in-js conversion later, but hopefully getting rid of global class names wherever possible will keep more global class names from popping up in new code and introducing unpredictable styling behavior.

Other changes:
- List and Button overview text updates
- Convert the last remaining SearchBox example using scss to css-in-js

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8508)

### CODE REVIEW

**Comment by khmakoto** [comment]
Closing and reopening to trigger new PR job on Dev Ops.

File: packages/office-ui-fabric-react/src/components/List/docs/ListOverview.md [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
+List breaks down the set of items passed in into pages. Only pages within a "materialized window" are actually rendered. As that window changes due to scroll events, pages that fall outside that window are removed, and their layout space is remembered and pushed into spacer elements. This gives the user the experience of browsing massive amounts of content but only using a small number of actual elements. This gives the browser much less layout to resolve, and gives React DOM diffing much less content to worry about.
 
-Note: although `onRenderCell` is an optional member of `IListProps`, if you do not provide the method, the `List` will still attempt to render the `name` property for each object within the provided `items` array.
+Note: if `onRenderCell` is not provided in `IListProps`, the List will attempt to render the `name` property for each object in the `items` array.
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/List/docs/ListOverview.md]: Unrelated to this PR, but I can't wait until we either make this type-safe OR remove the assumption completely...

File: packages/office-ui-fabric-react/src/components/List/examples/List.Basic.Example.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { TextField } from 'office-ui-fabric-react/lib/TextField';
 import { Image, ImageFit } from 'office-ui-fabric-react/lib/Image';
 import { Icon } from 'office-ui-fabric-react/lib/Icon';
 import { List } from 'office-ui-fabric-react/lib/List';
-import './List.Basic.Example.scss';
+import * as styles from './List.Basic.Example.scss';
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/List/examples/List.Basic.Example.tsx]: nice!

File: packages/office-ui-fabric-react/src/components/List/examples/List.Scrolling.Example.tsx [hunk 3]
```diff
@@ -123,14 +123,8 @@ export class ListScrollingExample extends React.Component<IListScrollingExampleP
...
-      <div className="ms-ListScrollingExample-itemCell" data-is-focusable={true}>
-        <div
-          className={css(
-            'ms-ListScrollingExample-itemContent',
-            index % 2 === 0 && 'ms-ListScrollingExample-itemContent-even',
-            index % 2 === 1 && 'ms-ListScrollingExample-itemContent-odd'
-          )}
-        >
+      <div data-is-focusable={true}>
+        <div className={css(styles.itemContent, index % 2 === 0 ? styles.itemContentEven : styles.itemContentOdd)}>
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/List/examples/List.Scrolling.Example.tsx]: yeesh.. this must predate even / odd psuedo selectors -.-''

**Pull request merged**